### PR TITLE
executor: refactor some methods of `Row` and `Chunk`

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Refactor some code of `Row`, which is prepared for introducing `Sel` to `Chunk`.

### What is changed and how it works?
1. hide some unsafe methods of `Chunk` which only used in tests;
2. refactor `Row` to use vectorized `Column` API;
3. add some methods for `Column`;

All changes are refactoring or new methods, so it's easy to review.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: util/chunk/column.go [hunk 1]
```diff
@@ -263,6 +263,31 @@ func (c *Column) Decimals() []types.MyDecimal {
...
+	return *(*float32)(unsafe.Pointer(&c.data[rowID*4]))
+}
+
+// GetFloat64 returns the float64 in the specific row.
+func (c *Column) GetFloat64(rowID int) float64 {
+	return *(*float64)(unsafe.Pointer(&c.data[rowID*8]))
+}
+
+// GetDecimal returns the decimal in the specific row.
+func (c *Column) GetDecimal(rowID int) types.MyDecimal {
```
Comment (zz-jason) [inline util/chunk/column.go]: how about only returning a reference of `MyDecimal`?

File: util/chunk/row.go [hunk 2]
```diff
@@ -110,27 +97,23 @@ func (r Row) getNameValue(colIdx int) (string, uint64) {
...
-	name, val := r.getNameValue(colIdx)
-	return types.Set{Name: name, Value: val}
+	return r.c.columns[colIdx].GetSet(r.idx)
 }
 
 // GetMyDecimal returns the MyDecimal value with the colIdx.
 func (r Row) GetMyDecimal(colIdx int) *types.MyDecimal {
-	col := r.c.columns[colIdx]
-	return (*types.MyDecimal)(unsafe.Pointer(&col.data[r.idx*types.MyDecimalStructSize]))
+	v := r.c.columns[colIdx].GetDecimal(r.idx)
```
Comment (coocood) [inline util/chunk/row.go]: Can we change `GetDecimal` to return pointer?

Comment (qw4990) [reply util/chunk/row.go]: fixed.

File: util/chunk/column.go (hunk 1 above)
Comment (qw4990) [reply util/chunk/column.go]: fixed, and it returns a pointer now.

**Comment by coocood** [comment]
LGTM

**Pull request merged**